cli/helpers.ts

```diff
@@ -0,0 +1,216 @@
+      this.backend = await getPort();
+    }
+    if (this.frontend === undefined) {
+      this.frontend = await getPort();
+    }
+    if (this.db === undefined) {
+      this.db = await getPort();
+    }
+  };
+}
+
+export const FRONTEND_REGEXP = /\NEXT_PUBLIC_GRAPHQL_URL=http:\/\/localhost:\d+\/graphql/;
+export const BACKEND_REGEXP = /\PORT=\d+/g;
+export const DATABASE_URL_REGEXP = /\DATABASE_URL=postgres:\/\/postgres:password@localhost\/postgres/;
+
+export class UnreachableCaseError extends Error {
+  constructor(val: never) {
+    super(`Unreachable case: ${JSON.stringify(val)}`);
+  }
+}
+
+export function copyTemplate(srcDir: string, targetDir: string) {
+  // Copy template, ignoring node_modules
+  try {
+    fse.copySync(srcDir, targetDir, {
+      filter: function (path) {
+        return !(path.indexOf("node_modules") > -1);
+      },
+    });
+    log(chalk.cyan(`Template copied to:\n  ${chalk.cyan.bold(targetDir)}`));
+  } catch {
+    throw new Error(
+      chalk.red(`Error copying template from ${srcDir} to ${targetDir}`)
+    );
+  }
+}
+
+export function updateTemplate(
+  appName: string,
+  targetDir: string,
+  ports: Ports
+) {
+  editReadme(appName, targetDir);
+  editPackageJson(appName, targetDir, Package.Frontend, ports);
+  editPackageJson(appName, targetDir, Package.Backend, ports);
+  editPackageJson(appName, targetDir, Package.TopLevel, ports);
+  editEnvFile(targetDir, Package.Frontend, ports);
+  editEnvFile(targetDir, Package.Backend, ports);
+  log(
+    chalk.cyan(
+      `Updated template with values: 
+  package name: ${chalk.cyan.bold(appName)}
+  frontend port: ${chalk.cyan.bold(ports.frontend)}
+  backend port: ${chalk.cyan.bold(ports.backend)}
+  database port: ${chalk.cyan.bold(ports.db)}
+`
+    )
+  );
+}
+
+export function editReadme(appName: string, targetDir: string) {
+  const filename = path.join(targetDir, `README.md`);
+
+  try {
+    const readmeContents = `#${appName}\n\nGenerated with create-originate-app`;
+    fs.writeFileSync(filename, readmeContents);
+    log(chalk.blue(`Prepared ${filename}`));
+  } catch (err) {
+    throw new Error(chalk.red(`Error preparing ${filename}\n${err}`));
+  }
+}
+
+export function editPackageJson(
+  appName: string,
+  targetDir: string,
+  packageName: Package,
+  ports: Ports
+) {
+  let filename!: string;
+  try {
+    switch (packageName) {
+      case Package.Frontend: {
+        // Update name package.json to use appName
+        filename = path.join(targetDir, `packages/${packageName}/package.json`);
+        let json = JSON.parse(fs.readFileSync(filename).toString());
+        json.scripts["name"] = `@${appName}/${packageName}`;
+        json.scripts[
+          "start:dev"
+        ] = `concurrently -k 'next dev -p ${ports.frontend}' 'yarn codegen:watch'`;
+        fs.writeFileSync(
+          filename,
+          prettier.format(JSON.stringify(json), {
+            semi: false,
+            parser: "json",
+          })
+        );
```

Comment:
Nice! This is indeed a situation where a parser (`JSON.parse`) is more appropriate than search-and-replace. I also like the use of prettier.

Can we centralize the prettier configuration? If we don't specify it, will prettier read from a `.prettierrc` file?